src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/reports/EnvelopeCountSummaryRepository.java

```diff
@@ -15,18 +15,26 @@
     @Query(
         nativeQuery = true,
         value = "SELECT\n"
-            + "  container,\n"
-            + "  date,\n"
+            + "  first_events.container AS container,\n"
+            + "  first_events.date AS date,\n"
             + "  count(*) AS received,\n"
-            + "  SUM(CASE WHEN event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE') THEN 1 ELSE 0 END) AS rejected\n"
+            + "  SUM(CASE WHEN rejection_events.id IS NOT NULL THEN 1 ELSE 0 END) AS rejected\n"
             + "FROM (\n"
             + "  SELECT DISTINCT on (container, zipfilename)\n"
-            + "  container, zipfilename, date(createdat), event\n"
+            + "    container, zipfilename, date(createdat), event\n"
             + "  FROM process_events\n"
             + "  ORDER BY container, zipfilename, date(createdat) ASC\n"
             + ") AS first_events\n"
-            + "GROUP BY container, date\n"
-            + "HAVING date = :date"
+            + "LEFT JOIN (\n"
+            + "  SELECT DISTINCT on (container, zipfilename)\n"
+            + "    id, container, zipfilename\n"
+            + "  FROM process_events\n"
+            + "  WHERE event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE')\n"
+            + ") AS rejection_events\n"
+            + "ON rejection_events.container = first_events.container\n"
+            + "AND rejection_events.zipfilename = first_events.zipfilename\n"
```

Comment:
not minding the spacings but looks like tthere's no need to expand to left join at all. how about moving ID to first subquery and but instead of ID just using case/when/then condition on `event` field like in the `where` clause of second subquery? then it will for sure be 0s and 1s in that collumn